src/App.tsx

```diff
@@ -15,49 +19,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'alphabetically',
+  ByLength = 'by length',
+  NoSorting = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): Good[] {
+  const preparedGoods = goods.map((good, index) => ({
+    name: good,
+    id: index + 1,
+  }));
+
+  if (sortField === SortType.Alphabetically
+    || sortField === SortType.ByLength
+  ) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabetically:
+          return good1.name.localeCompare(good2.name);
+
+        case SortType.ByLength:
+          return good1.name.length - good2.name.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.NoSorting);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer, sortField, isReversed,
+  );
+  const reset = () => {
+    setSortField(SortType.NoSorting);
+    setIsReversed(false);
+  };
+
+  const isSorted = (sortField || isReversed);
```

Comment:
But it could be `isReversed` too 🙄 
`isResetButtonVisible` or something similar can better describe this variable